### DISCUSSION TITLE
RabbitMQ Management UI Overflow policy validation failure

### DISCUSSION THREAD

**Discussion by zaeemarshad** [rabbitmq/rabbitmq-server#9635] [Other] [answered]
### Describe the bug

When adding a policy via the Management UI, the `overflow` policy is rejected with a validation failure. The same policy works via the `rabbitmqctl` cli.

<img width="1271" alt="Screenshot 2023-10-04 at 1 01 07 pm" src="https://github.com/rabbitmq/rabbitmq-server/assets/1113992/affd61f4-edc3-4974-9ef5-73f33e020093">

### Reproduction steps

1. Create a new policy via the Admin -> Policies page
2. Select `apply-to: Quorum Queues`
3. Set definition to `overflow: reject-publish`

### Expected behavior

Expect the policy to be created via the UI

**Comment by michaelklishin** [comment]
What version of RabbitMQ is used here?

**Reply by zaeemarshad** [reply]
Tested on 3.12.2 and 3.12.4

**Comment by michaelklishin** [comment]
So this value is not supposed by the operator policies and was removed from the form recently, which is why it is not in the documentation table. It could be supported but I don't know how then you can declare it via CLI tools. Both use the same validation code path.

**Reply by zaeemarshad** [reply]
That's not what I am seeing.

```
rmq-68f797597f-l64n4:/# rabbitmqctl set_policy qq-polic "^qq.*$" '{"max-length":100,"overflow":"reject-publish"}' --apply-to quorum_queues
Setting policy "qq-polic" for pattern "^qq.*$" to "{"max-length":100,"overflow":"reject-publish"}" with priority "0" for vhost "/" ...


rmq-68f797597f-l64n4:/# rabbitmqctl list_policies
Listing policies for vhost "/" ...
vhost	name	pattern	apply-to	definition	priority
/	qq-polic	^qq.*$	quorum_queues	{"max-length":100,"overflow":"reject-publish"}	0

rmq-68f797597f-l64n4:/# rabbitmqctl list_queues name arguments effective_policy_definition
Timeout: 60.0 seconds ...
Listing queues for vhost / ...
name	arguments	effective_policy_definition
qq-test	[{"x-queue-type","quorum"}]	[{<<"max-length">>, 100}, {<<"overflow">>, reject-publish}]
```

**Reply by zaeemarshad** [reply]
And more importantly, message rejection actually works. Am I missing something?

**Reply by mkuratczyk** [reply, marked as answer]
There are policies and there are operator policies. With the CLI, you defined a regular policy (`set_policy`, not `set_operator_policy`), which supports the `x-overflow` property, so everything works as expected here.

In the Management UI, you tried to define an Operator Policy, which currently does not support the `x-overflow` property, although perhaps it should.

https://www.rabbitmq.com/parameters.html#operator-policies

**Reply by michaelklishin** [reply]
@zaeemarshad you may want to consult

 * `rabbitmqctl help set_policy`
 * `rabbitmqctl help set_operator_policy`
 * The doc section on [operator policies](https://rabbitmq.com/parameters.html#operator-policies) and why they exist, how they are different from regular policies

**Reply by zaeemarshad** [reply]
🤦🏽 Thanks heaps!
